Bug 1207398 - Disable the backfill button on try repos

KWierso opened the pull request:
This is the easiest option I saw for this. It does mean the menu's tooltip is not completely correct (since it will still mention backfilling, even if that option is not present.

I guess the other options would be:
1) Also fix the tooltip for try repos.
2) On try, hide the entire menu and only show the old retrigger button.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/mozilla/treeherder/1102)
<!-- Reviewable:end -->

tojonmz commented:
> 1) Also fix the tooltip for try repos.

For what its worth, I favor option 1 :)

edmorley commented:
Yeah I think Jonathan's suggestion is a good one :-)

KWierso commented:
Any suggestions on the string?
________________________________
From: Ed Morley<mailto:notifications@github.com>
Sent: ‎11/‎1/‎2015 4:37 AM
To: mozilla/treeherder<mailto:treeherder@noreply.github.com>
Cc: KWierso<mailto:kwierso@gmail.com>
Subject: Re: [treeherder] Bug 1207398 - Hide the backfill button on try repos (#1102)

Yeah I think Jonathan's suggestion is a good one :-)

---
Reply to this email directly or view it on GitHub:
https://github.com/mozilla/treeherder/pull/1102#issuecomment-152823006

tojonmz commented:
> Any suggestions on the string?

Perhaps **Backfill not available for this repo** ?

...sort of the same wording the Log icon uses. I'm not sure if there are other examples around, but that was the first one that sprung to mind for me.
![lognotavailable](https://cloud.githubusercontent.com/assets/3660661/10870047/f57dac30-808e-11e5-872b-8bea57a5c209.jpg)

edmorley commented:
@KWierso does the above make sense? This is probably a quick-win in terms of finishing up and reducing the number of PRs we have open :-)

KWierso commented:
How about this? 

I put the `ng-class="disabled"` and the `ng-attr-title` on the <li> element so it gets applied. I put the `ng-disabled` on the <a> element itself because applying it to the <li> apparently disables the tooltip, too. This way, the hover effect gets applied to the <li>, the tooltip shows from the <li>, and the <a> element actually gets disabled so clicking it shouldn't trigger backfills. (I've only played with this by using ./web-server.js so I couldn't actually log in to see if it prevents the backfill from going through, but it should be correct...)

edmorley commented on ui/partials/main/thJobDetailsRetriggerMenu.html:
```diff
@@ -1,6 +1,7 @@
-  <a ng-click="backfillJob()">Backfill job</a>
+<li ng-class="currentRepo.repository_group.name == 'try' ? 'disabled' : ''"
+    ng-attr-title="{{currentRepo.repository_group.name == 'try' ? 'Backfilling not available in this repository' : ''}}">
+  <a ng-disabled="currentRepo.repository_group.name == 'try' ? 'disabled' : ''" ng-click="backfillJob()">Backfill job</a>
```
There's quite a bit of duplication here - I'd add a new function called something like `canBackfill` (similar to https://github.com/mozilla/treeherder/blob/4238fca43655caef566f0af3d1ff6d76a6dea41f/ui/plugins/controller.js#L356-L359) that you then refer to like happens here:
https://github.com/mozilla/treeherder/blob/2a9dbefa49d26d8d0ddc7c402c571691b1679770/ui/plugins/pluginpanel.html#L105

KWierso commented:
How about this? Flipped the logic for all of the checks since canBackfill() returns true if we aren't on try repos. Wrapped backfillJob with a check for canBackfill so it actually doesn't send the backfill request. (I guess ng-disabled doesn't do what I thought it did? Does it even do anything?)

KWierso commented:
Hmm, looks like ng-disabled would work for something like a default button action, but not for canceling out click events...

KWierso commented:
Rebased this around the newly landed bug 1207598, and added a second commit that moves the backfill button's title strings into controller.js because otherwise that line would be really long in the html file. Can put them back into the html file if you'd prefer, but other than that, I think this is ready.
